src/query/sql/src/planner/plans/window.rs

```diff
@@ -270,6 +257,14 @@ impl WindowFuncType {
                 default: default.map(Box::new),
                 return_type: Box::new(return_type),
             })),
+            "first_value" => Ok(WindowFuncType::FirstValue(FirstLastFunction {
```

Comment:
This function `get_general_window_func` is called by type_check and used for lag/lead only cause lag/lead 's return type depends on the default argument.
I see in the `transform_window.rs`, if the `frame_start >= frame_end` the value will be `Scalar::Null`, and the builder's data type is same as `first_value(col)`'s col data type, so if you need push null to the builder, you should change the return type in the `type_check.rs`.

Reply:
I think we cannot know whether it returning null in advance, because the range window frame will generate unpredictable frame.

Reply:
Oh, yes, i forget it.

Reply:
@gitccl is right. We can make the return type of all the window functions with arguments 
 and frame bound nullable.